### ISSUE TITLE
Activity email: Reviewer replies, information requests issues

### ISSUE THREAD

**Issue by wagnerand** [mozilla/addons-server#5081]
### Describe the problem and steps to reproduce it:

The activity email feature currently misses an important piece which makes it very difficult to work with and limits usefulness severely: Information requests are not picked up.

Actually the issue is two fold:
1) On AMO, reviewers cannot sent a _Reviewer reply_, only an _Information request_, which is not picked up by activity mail and therefore not sent to other involved reviewers (or members of the _Activity email CC_ group).
2) When a developer sends a _Developer reply_, a following reply by a reviewer is not necessarily an _Information request_. This is an issue because using the _Information request_ feature will set the add-on into a _Information requested_ state which has consequences, such as hiding it from the review queue.

What we should do instead is the following:
1) Replace the _Information request_ tab by a _Reviewer reply_ tab, and add a checkbox to that tab that indicates whether the reply is an _Information request_ or not. This gives reviewers the ability to send either an _Information request_ or a _Reviewer reply_.
2) Include _Information request_ emails in activity emails (and also remove the delivery to amo-editors).

**Comment by jvillalobos** [comment]
Why does there need to be a distinction between an info request an a reply? If a reviewer is sending something to the developer, it should be because there's still something pending from the developer.

**Comment by wagnerand** [comment]
Not every reviewer reply is a information request. Imagine the following scenario:

1. Developer sends a reply (for whatever reason) with some questions.
2. Reviewer replies and answers questions (currently using information request, although it's not really an information request).
3. Add-on is now marked as "Information needed", when it really doesn't need any information.

**Comment by jvillalobos** [comment]
After step 2, the add-on should either be reviewed by the reviewer, or it's pending something from the developer. I think adding the checkbox makes this more complicated than it should be.

**Comment by kewisch** [comment]
The upside to an "information request" is that these can be ignored in the list until the developer replies. If there is no such checkbox and everything is a developer reply, then reviewers will be looking at information requests multiple times until they are actually resolved.

**Comment by wagnerand** [comment]
> After step 2, the add-on should either be reviewed by the reviewer...

Unfortunately, the reality is that it's not working that way. There are plenty of examples only from a couple of weeks since we rolled this out.

Another example:

1. Reviewer rejects an add-on for some reason.
2. Developer sends a _Developer reply_ asking for clarification, more details etc.
3. Reviewer responds with explanations, further information etc. They are forced to use _Information request_. Now, while the _Information needed_ status on the add-on is of less importance in this case (because the add-on has been rejected), we still send out an email saying "A Reviewer requested further information ..." when they really don't.

**Comment by jvillalobos** [comment]
> Another scenario:

Yeah, I agree on naming the feature something more general than "Information request", since this is more about having an online chat between developers and reviewers. I also agree that those messages should be included in the activity email. What I think makes this more complicated than needs be is to have two different message types, so I think that should be dropped for now.

**Comment by wagnerand** [comment]
I considered that too. However, I do think that having a clear note indicating that an action is required is important.
As a follow up of this issue, I'd like to rewrite the need-info mail to make it similar to regular replies, or even use the same template and just add a note at the beginning.

**Comment by jvillalobos** [comment]
After some discussion on IRC, I'm still not sure about the checkbox and splitting replies and info requests. I suggest splitting it into a separate bug, and focus on the more critical issue here:

1. Rename the `Information request` tab in review pages to `Reviewer reply`. Reword the line that appears in the form from `This will request more information from the developer.` to `This will send a message to the developer.` 
2. Reword `emails/info.ltxt` from `A reviewer requested further information from you regarding version {{ number }} of your add-on {{ name }}` to `A reviewer sent you a message regarding version {{ number }} of your add-on {{ name }}`
3. Include these emails into activity emails, meaning they should be sent to the same recipients as other activity email.

@wagnerand What do you mean in the initial comment about  "remove the delivery to amo-editors"?

Setting P2 for these listed points (it's making activity email  muchless of a time savings for admins). Let's move the rest to another issue for a followup discussion.

**Comment by ValentinaPC** [comment]
Verified this on AMO-dev FF53(Win 7), and obtained the following results:
- "Request more info" tab was replaced by "Reviewer Reply" tab
- "Reviewer Reply" text line that appears in the form  was replace from "This will request more information from the developer." to "This will send a message to the developer"
- "Reviewer Reply" tab does not contain a checkbox to that tab that indicates whether the reply is an Information request or not
![rev_reply](https://cloud.githubusercontent.com/assets/15685960/25433922/e60b2c1e-2a92-11e7-8bb4-0e9988e3e06a.png)

- After adding a "Reviewer reply" in add-on review page is recorded that "More information requested"
![2017-04-26_1512](https://cloud.githubusercontent.com/assets/15685960/25433938/fd3f89ac-2a92-11e7-84e4-3433a11eb035.png)

- Same e-mail format is sent while `reviewer reply` versus `developer reply`
![mails](https://cloud.githubusercontent.com/assets/15685960/25433945/06037daa-2a93-11e7-8c81-59f38696db8b.png)

Are these the expected results?

**Comment by eviljeff** [comment]
TBH, a different template is supposed to be used now if the person sending the info request (or replying) is an editor.  Or more accurately, has access to send the message and isn't one of the add-on authors.
Which add-on was tested?  Were you one of the add-on authors too?

**Comment by ValentinaPC** [comment]
> Which add-on was tested? Were you one of the add-on authors too?

Add-on used: https://addons-dev.allizom.org/en-US/developers/addon/tlegacy-7/versions
Yes, I was also the author. 
The mails where sent the the people from Activity Mails CC, and were identical.


> "Reviewer Reply" tab does not contain a checkbox to that tab that indicates whether the reply is an Information request or not

Was this removed?

**Comment by eviljeff** [comment]
>> Which add-on was tested? Were you one of the add-on authors too?

> Add-on used: https://addons-dev.allizom.org/en-US/developers/addon/tlegacy-7/versions
Yes, I was also the author.
The mails where sent the the people from Activity Mails CC, and were identical.

Everyone gets the same template, because it only changes _why_ the message was sent, and that's the same for everyone.

(On prod you'd be blocked from reviewing your own add-on so you've hit an edge case).

If you remove yourself from the addon and send a reviewer message then a different template should be used.

>    "Reviewer Reply" tab does not contain a checkbox to that tab that indicates whether the reply is an Information request or not
> Was this removed?

Not removed, but the issue history above indicated this wasn't a requirement and a follow-up would be filed to add that if it's needed.

**Comment by ValentinaPC** [comment]
> After adding a "Reviewer reply" in add-on review page is recorded that "More information requested"

This one, too, is expected?

When I've replied from Review history - Dev Hub or from Reviewer Reply - Reviewer Tools for an add-on that wasn't mine things worked as expected.

**Comment by eviljeff** [comment]
I kept the 'more information requested' action even though it's named 'Reviewer Reply' in the UI.  If @wagnerand can log a new issue for his 'info requested' checkbox idea i'll change that in there so the 'info request' action is used when the checkbox is checked, etc.

**Comment by ValentinaPC** [comment]
We can mark this verified-fixed and follow-up the other details.
